diff --git a/src/terminal/c/terminal.zig b/src/terminal/c/terminal.zig
--- a/src/terminal/c/terminal.zig
+++ b/src/terminal/c/terminal.zig
@@ -98,7 +98,7 @@ const ResponseHandler = struct {
             .insert_lines => self.terminal.insertLines(value),
             .insert_blanks => self.terminal.insertBlanks(value),
             .delete_lines => self.terminal.deleteLines(value),
-            .scroll_up => self.terminal.scrollUp(value),
+            .scroll_up => try self.terminal.scrollUp(value),
             .scroll_down => self.terminal.scrollDown(value),
             .horizontal_tab => try self.horizontalTab(value),
             .horizontal_tab_back => try self.horizontalTabBack(value),
